ibis/common/grounds.py

```diff
@@ -96,6 +96,12 @@ def __create__(cls, *args, **kwargs) -> Annotable:
         kwargs = cls.__signature__.validate(*args, **kwargs)
         return super().__create__(**kwargs)
 
+    @classmethod
+    def __recreate__(cls, kwargs) -> Annotable:
```

Comment:
I couldn't come up with a better name, it's primarily used in `__reduce__`.

It is basically a variant of `__create__` without the flexible signature handling. For a signature like `Signature(a, b, c, *args, **kwargs)` the following two calls create the same object:

```py
from ibis.common.validators import instance_of
from ibis.common.annotations import varargs, varkwds
from ibis.common.grounds import Annotable

class MyClass(Annotable):
    a = instance_of(int)
    b = instance_of(int)
    args = varargs(instance_of(int))
    kwargs = varkwds(instance_of(int))

MyClass.__create__(1, 2, 3, 4, e=5, f=6)
MyClass.__recreate__(a=1, b=1, args=(3, 4), kwargs={'e': 5, 'f': 6})
```

I was thinking of calling it `def new(cls, kwargs)` to better align with `copy()` which also only supports direct field/attribute passing using kwargs.

Reply:
```py
MyClass(1, 2, 3, 4, e=5, f=6)
MyClass.new(a=1, b=1, args=(3, 4), kwargs={'e': 5, 'f': 6})
```

The latter should be more performant since it bypasses https://github.com/python/cpython/blob/3.11/Lib/inspect.py#L3075

Reply:
How about `__strict_create__`?

Reply:
Or even `__create_impl__` might be okay.

Reply:
This is the API we need to use in the tree substitute/replace functions, so perhaps this should be public. 

Another approach could be the following:

```py
class Annotable:
    ...

    @classmethod
    def from_dict(cls, dct):
        # bypass signature binding by requiring keyword arguments only
        kwargs = cls.__signature__.validate_nobind(**kwargs)
        return super().__create__(**kwargs)

    def to_dict(self):
        return {k: getattr(self, k) for k in self.__attributes__}
```

Reply:
Though `from_dict` and `to_dict` wouldn't be symmetric in the example above, since the latter would return with the generated state/attributes as well while the former would accept only the argument kind of attributes.

Reply:
We can leave it private for now IMO